src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/AtMostOneFromBodyAttribute.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+using Microsoft.AspNetCore.App.Analyzers.Infrastructure;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteHandlers;
+
+using WellKnownType = WellKnownTypeData.WellKnownType;
+
+public partial class RouteHandlerAnalyzer : DiagnosticAnalyzer
+{
+    private static void AtMostOneFromBodyAttribute(
+        in OperationAnalysisContext context,
+        WellKnownTypes wellKnownTypes,
+        IMethodSymbol methodSymbol)
+    {
+        var fromBodyMetadataInterfaceType = wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromBodyMetadata);
+        var asParametersAttributeType = wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_AsParametersAttribute);
+
+        var asParametersDecoratedParameters = methodSymbol.Parameters.Where(p => p.HasAttribute(asParametersAttributeType));
+
+        foreach (var asParameterDecoratedParameter in asParametersDecoratedParameters)
+        {
+            var fromBodyMetadataInterfaceMembers = asParameterDecoratedParameter.Type.GetMembers().Where(
+                m => m.HasAttributeImplementingInterface(fromBodyMetadataInterfaceType)
+                );
+
+            if (fromBodyMetadataInterfaceMembers.Count() >= 2)
+            {
+                ReportDiagnostics(context, fromBodyMetadataInterfaceMembers);
+            }
+        }
+
+        var fromBodyMetadataInterfaceParameters = methodSymbol.Parameters.Where(p => p.HasAttributeImplementingInterface(fromBodyMetadataInterfaceType));
+
+        if (fromBodyMetadataInterfaceParameters.Count() >= 2)
+        {
+            ReportDiagnostics(context, fromBodyMetadataInterfaceParameters);
```

Comment:
This wouldn't fail if there's a single `[FromBody]` parameter and then another `[AsParameter]` parameter with a single `[FromBody]` property, right? Or multiple `[AsParameter]` parameters with a single `[FromBody]` property each? All of these should fail. Can we add test cases for these?